sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ExpressionEvaluationSuite.scala

```diff
@@ -59,6 +60,10 @@ class ExpressionEvaluationBaseSuite extends FunSuite {
 
 class ExpressionEvaluationSuite extends ExpressionEvaluationBaseSuite {
 
+  def create_row(values: Any*): Row = {
+    new GenericRow(values.map(ScalaReflection.convertToCatalyst).toArray)
```

Comment:
I see.  Perhaps that is actually the way that we should handle inbound conversions for data coming from users, instead of doing it on inbound RDDs.  If we make `Row.apply` do the conversion then we always have converted data from users.  Low level APIs construct rows or mutable rows directly to avoid the conversion.